### PR TITLE
Image comfrimation for food box deliveries 

### PR DESCRIPTION
**Pull request by thirdeyeclub**
A new storage file in Firebase was created to handle and better sort the images for delivery confirmation!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
     "dot-object": "^2.1.3",
     "express": "^4.17.1",
     "firebase": "^7.14.0",
+    "firebase-functions": "^3.6.1",
```
Comment (qdozaq) [inline package.json:24]: Do we need this?

File: src/app/component/MissionComponent/MissionDetailsCard/MissionDetailsCard.jsx [hunk 2]
```diff
@@ -160,4 +187,4 @@ MissionDetailsCard.propTypes = {
   mission: PropTypes.object.isRequired,
 };
 
-export default MissionDetailsCard;
+export default withRouter(withFirestore(MissionDetailsCard));
```
Comment (qdozaq) [inline src/app/component/MissionComponent/MissionDetailsCard/MissionDetailsCard.jsx:190]: Doesn't look like you're using the router or firestore as a `prop` for this component so can get rid of the HOCs

File: src/app/component/MissionComponent/MissionDetailsCard/MissionDetailsCard.jsx [hunk 3]
```diff
@@ -70,12 +72,37 @@ const Row = ({ children, Icon, label }) => {
  * @component
  */
 const MissionDetailsCard = ({ mission, photoDisabled }) => {
-  photoDisabled = true;
+  const snackbarContext = useContext(Snackbar.Context.SnackbarContext);
+  const firebase = getFirebase();
```
Comment (qdozaq) [inline src/app/component/MissionComponent/MissionDetailsCard/MissionDetailsCard.jsx:76]: Should use `useFirebase` here.

`getFirebase` isn't a hook and therefore can't update react fire something changes

File: src/app/component/MissionComponent/MissionDetailsCard/MissionDetailsCard.jsx [hunk 4]
```diff
@@ -70,12 +72,37 @@ const Row = ({ children, Icon, label }) => {
...
+  }
+
+  async function handleImageChosen(file) {
     setFile(file);
+    if (file) {
+      const uploadTask = storage
+        .ref(`confirmed_delivery/${file.name}`)
+        .put(file)
+        .then((data) => {
+          return data.ref.getDownloadURL();
```
Comment (qdozaq) [inline src/app/component/MissionComponent/MissionDetailsCard/MissionDetailsCard.jsx:94]: How do we access this image now?
does this URL need to get added to the mission?

File: package.json (hunk 1 above)
Comment (mat10tng) [reply package.json:24]: yes, for the running of emulator

Comment (utunga) [reply package.json:24]: oops i also added it in another pr per request from Tuan thats gonna be a potential clash hopefully this PR can land first though

File: src/app/component/MissionComponent/MissionDetailsCard/MissionDetailsCard.jsx (hunk 4 above)
Comment (thirdeyeclub) [reply src/app/component/MissionComponent/MissionDetailsCard/MissionDetailsCard.jsx:94]: Yes I have been trying to add the URL to the mission, but it wont update the firebase for some reason

File: src/app/component/MissionComponent/MissionDetailsCard/MissionDetailsCard.jsx [hunk 5]
```diff
@@ -72,27 +74,39 @@ const Row = ({ children, Icon, label }) => {
...
   } else {
     photoDisabled = true;
   }
 
+  function addImage(img) {
+    const imgURL = Object.values(img)[1];
+    img
+      ? console.log(imgURL) && Mission.update(mission.uid, { deliveryConfirmationImage: imgURL })
+      : snackbarContext.show({
+          message: `Unable to add image to Mission ${mission.uid}: ${error.message}`,
```
Comment (jwu910) [inline src/app/component/MissionComponent/MissionDetailsCard/MissionDetailsCard.jsx]: If you plan on showing an error, this add image should happen in a try-catch so you have an error object to consume

File: src/app/component/MissionComponent/MissionDetailsCard/MissionDetailsCard.jsx [hunk 6]
```diff
@@ -72,27 +74,39 @@ const Row = ({ children, Icon, label }) => {
...
   if (mission.status === "started") {
     photoDisabled = false;
   } else {
     photoDisabled = true;
   }
 
+  function addImage(img) {
+    const imgURL = Object.values(img)[1];
+    img
+      ? console.log(imgURL) && Mission.update(mission.uid, { deliveryConfirmationImage: imgURL })
```
Comment (jwu910) [inline src/app/component/MissionComponent/MissionDetailsCard/MissionDetailsCard.jsx:92]: Extra console log

File: src/app/page/Home/Home.js [hunk 7]
```diff
@@ -381,7 +381,7 @@ const HomePage = ({ auth, history, profile }) => {
     if (!auth.uid) {
       return [];
     }
-    const id = org.uid;
+    const id = "1";
```
Comment (jwu910) [inline src/app/page/Home/Home.js:384]: Is this an accidental commit?

File: src/app/component/ImageUpload/ImageUpload.js [hunk 8]
```diff
@@ -14,7 +14,7 @@ import { Actions, Container, StyledIcon } from "./ImageUpload.style";
 export function ImageUpload({ styles, getFile = () => null, ...props }) {
   const fileUpload = useRef();
   const [file, setFile] = useState("");
-  const [preview, setPreview] = useState(null);
+  const [preview, setPreview] = useState(props.dCI);
```
Comment (jwu910) [inline src/app/component/ImageUpload/ImageUpload.js:17]: If we're pulling this from props, can we add it to the propTypes?

File: src/app/model/schema.ts [hunk 9]
```diff
@@ -208,9 +208,9 @@ export interface MissionInterface {
 
   deliveryWindow: TimeWindow | null;
   deliveryLocation: Location; // default to recipient location
-  deliveryConfirmationImage: ImageUrl;
-  deliveryType: "curbside" | "delivery";
+  deliveryConfirmationImage: any;
   deliveryNotes: string;
+  deliveryType: string;
```
Comment (jwu910) [inline src/app/model/schema.ts:213]: "curbside" and "delivery" changes in the scope of this ticket?

File: src/app/component/MissionComponent/MissionDetailsCard/MissionDetailsCard.jsx [hunk 10]
```diff
@@ -147,7 +188,11 @@ const MissionDetailsCard = ({ mission, photoDisabled }) => {
               it.
             </Row>
             <Grid item container justify="center">
-              <ImageUpload withoutTwoBtns getFile={handleImageChosen} />
+              <ImageUpload
+                withoutTwoBtns
+                getFile={handleImageChosen}
+                dCI={mission.deliveryConfirmationImage}
```
Comment (jwu910) [inline src/app/component/MissionComponent/MissionDetailsCard/MissionDetailsCard.jsx:194]: can dCI be more descriptive?

File: src/app/component/MissionComponent/MissionDetailsCard/MissionDetailsCard.jsx [hunk 11]
```diff
@@ -15,6 +15,9 @@ import ImageUpload from "../../ImageUpload";
 import FoodBoxIcon from "../../icons/FoodBoxIcon";
 import { H2 } from "../../Typography";
 import styled from "styled-components";
+import { useFirebase } from "react-redux-firebase";
+import { set } from "date-fns";
```
Comment (jwu910) [inline src/app/component/MissionComponent/MissionDetailsCard/MissionDetailsCard.jsx:19]: Unused import

File: src/app/page/Home/Home.js (hunk 7 above)
Comment (thirdeyeclub) [reply src/app/page/Home/Home.js:384]: Yes it was an accident

File: src/app/component/MissionComponent/MissionDetailsCard/MissionDetailsCard.jsx (hunk 6 above)
Comment (thirdeyeclub) [reply src/app/component/MissionComponent/MissionDetailsCard/MissionDetailsCard.jsx:92]: Yes that is there to show that the mission update is broken, it takes the url but wont update mission

File: src/app/model/schema.ts (hunk 9 above)
Comment (thirdeyeclub) [reply src/app/model/schema.ts:213]: that was added when I pulled from master and broke everything unless I added line 213 to mission interface

**Pull request merged**